### PR TITLE
Ignore placeholder selectors; don't identify them as standard type selectors.

### LINKED ISSUE #1334: selector-type-no-unknown doesn't recognize Sass placeholder selector
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`selector-type-no-unknown` rule doesn't recognize [Sass placeholder selector](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#placeholder_selectors_) (`%foo`).

> Which rule, if any, is this issue related to?

`selector-type-no-unknown`

> What CSS is needed to reproduce this issue?

```sass
%icon-loading {
  ...
}
```

> Which version of stylelint are you using?

`6.4.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

NPM script with `stylelint --syntax scss 'app/assets/stylesheets/scss/**/*.scss'`.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to Sass syntax.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warning was flagged:

```shell
738:1   ⚠  Unexpected unknown type selector "%icon-loading"   selector-type-no-unknown
```

### PR DESCRIPTION
**Pull request by mehcode**
Fixes #1334

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # 6.4.2
 
 - Fixed: `selector-pseudo-class-case`, `selector-pseudo-class-no-unknown`, `selector-pseudo-element-case`, `selector-pseudo-element-no-unknown` rules now ignore SCSS variable interpolation.
-- Fixed: `selector-type-no-unknown` now ignores nested selectors and keyframe selectors.
+- Fixed: `selector-type-no-unknown` now ignores nested selectors, keyframe selectors, and placeholder selectors.
+- Fixed: `selector-type-case` now ignores placeholder selectors.
```
Comment (davidtheclark) [inline CHANGELOG.md]: Because 6.4.2 has been released, you'll need to create a new section in the Changelog titled `# Head` and add your fixes there.

**Comment by davidtheclark** [comment]
Thanks! Just a tweak to the changelog necessary, then I'm good with it.

**Comment by jeddy3** [comment]
Perhaps this should be dealt with within [`isStandardSelector()`](https://github.com/stylelint/stylelint/blob/master/src/utils/isStandardSelector.js), rather than `isStandardTypeSelector()`?

**Comment by mehcode** [comment]
@jeddy3 Was about to say something about compounded selectors but a placeholder selector is just `%name` with nothing after or before. That would work and is a more global no vs just blocking on a tag name check. I like it.

**Comment by jeddy3** [comment]
@mehcode Thanks for this!

The commit history in this PR got a little messed up, so I fixed it up in https://github.com/stylelint/stylelint/pull/1338

**Pull request closed without merging**